src/App.tsx

```diff
@@ -1,223 +1,164 @@
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case FilterTypes.active:
+        return !todo.completed;
+
+      case FilterTypes.completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos(user?.id || 0);
+
+        setTodos(loadedTodos);
+      } catch {
+        setError(ErrorTypes.LOAD);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => setError(null), 3000);
+    }
+  }, [error]);
+
+  const addNewTodo = async () => {
+    setIsAdding(true);
+
+    if (!todoToAdd.length) {
+      setError(ErrorTypes.LOAD);
+      setTimeout(() => setError(null), 3000);
+      setTodoToAdd('');
+      setIsAdding(false);
+
+      return;
+    }
+
+    if (user) {
+      try {
+        const newAPITodo = await addTodo(todoToAdd, user.id);
+
+        setTodos(currTodos => [...currTodos, newAPITodo]);
+      } catch {
+        setError(ErrorTypes.ADD);
+      }
+    }
+
+    setTodoToAdd('');
+    setIsAdding(false);
+  };
+
+  const handleDeleteTodo = useCallback(async (todoId: number) => {
+    setDeletedIds([todoId]);
+
+    if (user) {
+      try {
+        await deleteTodo(todoId);
+        setTodos(prev => (
+          prev.filter(item => item.id !== todoId)
+        ));
+      } catch {
+        setError(ErrorTypes.DELETE);
+      }
+    }
+  }, [user]);
+
+  const handleDeleteAllTodos = async () => {
+    try {
+      await Promise.all(todos.map(async (todo) => {
+        if (todo.completed) {
+          await deleteTodo(todo.id);
+        }
+
+        return todo;
+      }));
+
+      setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+    } catch {
+      setError(ErrorTypes.DELETEALL);
+    }
+  };
+
+  const handleFilterType = useCallback((type: FilterTypes) => {
+    setFilterType(type);
   }, []);
 
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+        <TodoField
+          todos={todos}
+          newTodo={newTodoField}
+          isAdding={isAdding}
+          addNewTodo={() => addNewTodo()}
```

Comment:
you don't have arguments, so it's not necessary to  call it inside another function